build/scripts/check-digests.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+cd "$SCRIPT_DIR" || { echo "Failed to change directory to $SCRIPT_DIR"; exit 1; }
+DIGESTS_DIR="../../target/digests"
+
+# Check if the digests directory exists
+if [ ! -d "$DIGESTS_DIR" ]; then
+    echo "Digests directory not found: $DIGESTS_DIR"
+    exit 1
+fi
+
+# Iterate over each digest file in the digests directory
+for DIGEST_FILE in "$DIGESTS_DIR"/*; do
+    # Skip if it's not a file
+    if [ ! -f "$DIGEST_FILE" ]; then
+        continue
+    fi
+
+    FILENAME=$(basename "$DIGEST_FILE" .DIGESTS)
```

Comment:
```suggestion
    FILENAME="$(basename "$DIGEST_FILE" .DIGESTS)"
```
Output from command substitution is split on whitespace otherwise. Maybe not relevant here, but good practice.